File: src/main/java/cz/nitramek/messaging/Communicator.java
package cz.nitramek.messaging;

import cz.nitramek.messaging.network.MessageCallback;

import java.io.File;
import java.net.InetSocketAddress;

/**
 * It is expected if you have an instance of coomunicator that you keep it.
 */
public interface Communicator {

    void sendMessage(Command command, InetSocketAddress recipient, String data, File file);

    void registerMessageCallback(Command command, MessageCallback callback);

    void stop();

    void start();
}

File: src/main/java/cz/nitramek/messaging/Communicator.kt
package cz.nitramek.messaging

import cz.nitramek.messaging.message.Message
import cz.nitramek.messaging.message.MessageHandler
import cz.nitramek.messaging.message.MessageHeader

import java.net.InetSocketAddress

/**
 * It is expected if you have an instance of coomunicator that you keep it.
 */
interface Communicator {

    /**
     * Sends message to address, if the message acked it will add the address to addressList
     */
    fun sendMessage(message: Message, address: InetSocketAddress, acked: Boolean, priority: Boolean)

    fun sendMessage(message: String, address: InetSocketAddress, acked: Boolean, priority: Boolean)

    fun addMessageHandler(handler: MessageHandler)

    fun stop()

    fun start()

    fun respondAdress(): InetSocketAddress

    val addressBook: Map<InetSocketAddress, Int>
    fun addNewAgentAddress(header: MessageHeader)

    fun sendImmediadly(message: Message, address: InetSocketAddress, acked: Boolean)
}
